### PR TITLE
Basic multi-user UI. r=fabrice

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

File: components/thinkerbell/src/manager.rs [hunk 1]
```diff
@@ -76,7 +76,7 @@ impl<Env, T> ScriptManager<Env, T>
             id          TEXT NOT NULL PRIMARY KEY,
             source      TEXT NOT NULL,
             is_enabled  BOOL NOT NULL DEFAULT 1,
-            owner       INTEGER NOT NULL DEFAULT -1
+            owner       TEXT
```
Comment (fabricedesre) [inline components/thinkerbell/src/manager.rs:79]: We don't care yet, but we really need to plan for db updates...

File: src/adapters/webpush/mod.rs [hunk 2]
```diff
@@ -42,7 +42,7 @@ static ADAPTER_NAME: &'static str = "WebPush adapter (built-in)";
 static ADAPTER_VENDOR: &'static str = "team@link.mozilla.org";
 static ADAPTER_VERSION: [u32;4] = [0, 0, 0, 0];
 // This user identifier will be used when authentication is disabled.
-static NO_AUTH_USER_ID: i32 = -1;
+static NO_AUTH_USER_ID: &'static str = "-1";
```
Comment (fabricedesre) [inline src/adapters/webpush/mod.rs]: this looks pretty hackish... We should stick with None instead.

**Comment by fabricedesre** [comment]
There are a few things I'd like to see improve:
- webpush using None instead of "-1"
- more comments in the js frontend code. I glanced over, and there are parts where tbh I'm not sure what's happening.

Follow ups to keep in mind:
- moving the User type in foxbox_users and use it elsewhere.
- clean db upgrades (it's not limited to webpush though).
- l10n for the frontend.

**Comment by ferjm** [comment]
> moving the User type in foxbox_users and use it elsewhere.

Actually, I'd prefer to make this simply an Option<String> so we don't create unnecessary dependencies (i.e. taxonomy with users).

**Comment by ferjm** [comment]
@fabricedesre could you take another look at this, please?

File: src/adapters/webpush/db.rs [hunk 3]
```diff
@@ -159,64 +168,65 @@ pub fn remove_test_db() {
...
     before_each {
+        use foxbox_taxonomy::api::User;
         let db = WebPushDb::new(&get_db_environment());
     }
 
     it "should manage subscription correctly" {
         use super::super::Subscription;
 
-        let subs0 = db.get_subscriptions(1).unwrap();
+        let subs0 = db.get_subscriptions(&User::Id("1".to_owned())).unwrap();
```
Comment (fabricedesre) [inline src/adapters/webpush/db.rs]: nit: you're using String::from("1") everywhere else.

**Comment by fabricedesre** [comment]
r=me thanks!

**Pull request merged**